CMakeLists.txt

```diff
@@ -4,8 +4,8 @@ project(gtda_bindings LANGUAGES CXX)
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/gtda/externals/pybind11)
 set(BINDINGS_DIR "gtda/externals/bindings")
 
-find_package(Boost 1.56 REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
+find_package(Boost 1.57 REQUIRED)
+include_directories(${Boost_INCLUDE_DIRS})
```

Comment:
Is the plural `DIRS` important here?

Reply:
Unfortunately yes ...
From the official [documentation]8https://cmake.org/cmake/help/latest/module/FindBoost.html):

> Boost_INCLUDE_DIR         - Directory containing Boost headers
> Boost_INCLUDE_DIRS     - Boost include directories

But I'm not sure why we should specify `Boost include directories` instead of `Directory containing Boost headers`.
Can you elaborate a bit more @gtauzin ?

Reply:
The reason why I changed it was simply because I tried to install the library in dev mode on my linux laptop and it did not work.

My understanding is that we need to include the directories as they are defined when boost is found and that is the `DIRS` one.

Reply:
I'm not really convinced,did you clear CMake cache entries after upgrading your boost version ?
Because from what I understand in the [documentation](https://cmake.org/cmake/help/v3.9/module/FindBoost.html), `Boost_INCLUDE_DIR` is saved persistently in CMake cache entries and `Boost_INCLUDE_DIRS` is not. 
From my observations, `Boost_INCLUDE_DIRS` and `Boost_INCLUDE_DIR` have the same value, but from what I understand in the documentation it could happen that they are not.

I would rethink twice before changing `include_directories(${Boost_INCLUDE_DIR})`, maybe ask community (SO) about this issue ?